services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -184,7 +189,11 @@ const getLastBlock = async () => {
 const isQueryFromIndex = params => {
 	const paramProps = Object.getOwnPropertyNames(params);
 
-	const isDirectQuery = ['id', 'height', 'heightBetween'].some(prop => paramProps.includes(prop));
+	const directQueryParams = ['id', 'height', 'heightBetween'];
+	const defaultQueryParams = ['limit', 'offset', 'sort'];
+	const isDirectQuery = directQueryParams.some(prop => paramProps.includes(prop))
+		&& (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
+		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```

Comment:
Please consider commenting on what each part of the logic aims to achieve.
```suggestion
	// At least one of the 'directQueryParams' is present within the request params
	// Only one of the 'directQueryParams' is present within the request params
	// All the request params are either 'directQueryParams' or 'defaultQueryParams'
	const isDirectQuery = directQueryParams.some(prop => paramProps.includes(prop))
		&& (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```

Also, please test if the following works:
```suggestion
	// For 'isDirectQuery' to be 'true', the request params should contain exactly one of 'directQueryParams'
	// and all of them must be contained within 'directQueryParams' or 'defaultQueryParams'
	const isDirectQuery = (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```

Reply:
I have tested the suggestion and it worked as expected.